ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Byron/Aux.hs

```diff
@@ -0,0 +1,545 @@
+applyScheduledDelegations update (Delegation.Map del) =
+    Delegation.Map $ Foldable.foldl' (flip applyOne) del update
+  where
+    applyOne :: D.Sched.ScheduledDelegation
+             -> Bimap CC.KeyHash CC.KeyHash
+             -> Bimap CC.KeyHash CC.KeyHash
+    applyOne x = Bimap.insert (D.Sched.sdDelegator x)
+                              (D.Sched.sdDelegate  x)
+
+{-------------------------------------------------------------------------------
+  Applying transactions
+-------------------------------------------------------------------------------}
+
+mkUtxoEnvironment :: Gen.Config
+                  -> CC.ChainValidationState
+                  -> Utxo.Environment
+mkUtxoEnvironment cfg state = Utxo.Environment {
+      Utxo.protocolMagic      = protocolMagic
+    , Utxo.protocolParameters = U.Iface.adoptedProtocolParameters updateState
+    , Utxo.utxoConfiguration  = Gen.configUTxOConfiguration cfg
+    }
+  where
+    protocolMagic = reAnnotateMagic (Gen.configProtocolMagic cfg)
+    updateState   = CC.cvsUpdateState state
+
+mkDelegationEnvironment :: Gen.Config
+                        -> CC.ChainValidationState
+                        -> D.Iface.Environment
+mkDelegationEnvironment cfg state = D.Iface.Environment {
+      D.Iface.protocolMagic     = getAProtocolMagicId protocolMagic
+    , D.Iface.allowedDelegators = allowedDelegators cfg
+    , D.Iface.k                 = k
+    , D.Iface.currentEpoch      = currentEpoch
+    , D.Iface.currentSlot       = currentSlot
+    }
+  where
+    k             = Gen.configK cfg
+    protocolMagic = reAnnotateMagic (Gen.configProtocolMagic cfg)
+    currentSlot   = CC.cvsLastSlot state
+    currentEpoch  = CC.slotNumberEpoch (Gen.configEpochSlots cfg) currentSlot
+
+mkUpdateEnvironment :: Gen.Config
+                    -> CC.ChainValidationState
+                    -> U.Iface.Environment
+mkUpdateEnvironment cfg state = U.Iface.Environment {
+      U.Iface.protocolMagic = getAProtocolMagicId protocolMagic
+    , U.Iface.k             = k
+    , U.Iface.currentSlot   = currentSlot
+    , U.Iface.numGenKeys    = numGenKeys
+    , U.Iface.delegationMap = delegationMap
+    }
+  where
+    k             = Gen.configK cfg
+    protocolMagic = reAnnotateMagic (Gen.configProtocolMagic cfg)
+    currentSlot   = CC.cvsLastSlot state
+    numGenKeys    = toNumGenKeys $ Set.size (allowedDelegators cfg)
+    delegationMap = getDelegationMap state
+
+    toNumGenKeys :: Int -> Word8
+    toNumGenKeys n
+      | n > fromIntegral (maxBound :: Word8) = error $
+        "toNumGenKeys: Too many genesis keys"
+      | otherwise = fromIntegral n
+
+applyTxAux :: MonadError Utxo.UTxOValidationError m
+           => CC.ValidationMode
+           -> Gen.Config
+           -> Utxo.ATxAux ByteString
+           -> CC.ChainValidationState -> m CC.ChainValidationState
+applyTxAux validationMode cfg tx state =
+    (`setUTxO` state) <$>
+      runReaderT (Utxo.updateUTxO utxoEnv utxo [tx]) validationMode
```

Comment:
Generalize signature to allow for list of transactions.
Use same `runReaderT` style as above.